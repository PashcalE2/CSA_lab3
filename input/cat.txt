(1, 'A'), (2, 'l'), (3, 'i'), (5, 'c'), (6, 'e'), (7, ' '), (8, '1'), (9, '9'), (12, '9'), (13, '9'), (15, '!'), (23, 32)