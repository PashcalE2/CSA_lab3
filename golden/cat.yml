in_source: |-
  ,[.,]
in_stdin: |-
  foo
out_code: |-
  [{"index": 0, "opcode": "input", "term": [1, 1, ","]},
   {"index": 4, "opcode": "jz", "arg": 5, "term": [1, 2, "["]},
   {"index": 2, "opcode": "print", "term": [1, 3, "."]},
   {"index": 3, "opcode": "input", "term": [1, 4, ","]},
   {"index": 4, "opcode": "jmp", "arg": 1, "term": [1, 5, "]"]},
   {"opcode": "halt"}]
out_stdout: |
  source LoC: 1 code instr: 6
  ============================================================
  foo
  instr_counter:  11 ticks: 21
out_log: |
  DEBUG   machine:simulation    TICK:   0 PC:   0 ADDR:   0 MEM_OUT: 0 ACC: 0 	input  (','@1:1)
  DEBUG   machine:signal_wr     input: 'f'
  DEBUG   machine:simulation    TICK:   2 PC:   1 ADDR:   0 MEM_OUT: 102 ACC: 0 	jz 5  ('['@1:2)
  DEBUG   machine:simulation    TICK:   4 PC:   2 ADDR:   0 MEM_OUT: 102 ACC: 102 	print  ('.'@1:3)
  DEBUG   machine:signal_output output: '' << 'f'
  DEBUG   machine:simulation    TICK:   6 PC:   3 ADDR:   0 MEM_OUT: 102 ACC: 102 	input  (','@1:4)
  DEBUG   machine:signal_wr     input: 'o'
  DEBUG   machine:simulation    TICK:   8 PC:   4 ADDR:   0 MEM_OUT: 111 ACC: 102 	jmp 1  (']'@1:5)
  DEBUG   machine:simulation    TICK:   9 PC:   1 ADDR:   0 MEM_OUT: 111 ACC: 102 	jz 5  ('['@1:2)
  DEBUG   machine:simulation    TICK:  11 PC:   2 ADDR:   0 MEM_OUT: 111 ACC: 111 	print  ('.'@1:3)
  DEBUG   machine:signal_output output: 'f' << 'o'
  DEBUG   machine:simulation    TICK:  13 PC:   3 ADDR:   0 MEM_OUT: 111 ACC: 111 	input  (','@1:4)
  DEBUG   machine:signal_wr     input: 'o'
  DEBUG   machine:simulation    TICK:  15 PC:   4 ADDR:   0 MEM_OUT: 111 ACC: 111 	jmp 1  (']'@1:5)
  DEBUG   machine:simulation    TICK:  16 PC:   1 ADDR:   0 MEM_OUT: 111 ACC: 111 	jz 5  ('['@1:2)
  DEBUG   machine:simulation    TICK:  18 PC:   2 ADDR:   0 MEM_OUT: 111 ACC: 111 	print  ('.'@1:3)
  DEBUG   machine:signal_output output: 'fo' << 'o'
  DEBUG   machine:simulation    TICK:  20 PC:   3 ADDR:   0 MEM_OUT: 111 ACC: 111 	input  (','@1:4)
  WARNING machine:simulation    Input buffer is empty!
  INFO    machine:simulation    output_buffer: 'foo'
